sources/core/Xenko.Core/Threading/ThreadPool.cs

```diff
@@ -19,109 +20,71 @@ internal class ThreadPool
         public static readonly ThreadPool Instance = new ThreadPool();
 
         private readonly int maxThreadCount = Environment.ProcessorCount + 2;
-        //private readonly int maxThreadCount = Environment.ProcessorCount * 2;
-        private readonly List<Task> workers = new List<Task>();
         private readonly Queue<Action> workItems = new Queue<Action>();
         private readonly ManualResetEvent workAvailable = new ManualResetEvent(false);
 
-        private SpinLock spinLock = new SpinLock();
-        private int activeThreadCount;
+        private object _lock = new object();
+        volatile private int activeThreadCount;
```

Comment:
Modifier order: `private volatile`. But anyway, `Interlocked` and `volatile` should not be combined, either use one or the other but not both.